codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/UnionGenerator.kt

```diff
@@ -26,6 +28,12 @@ fun CodegenTarget.renderUnknownVariant() = when (this) {
     CodegenTarget.CLIENT -> true
 }
 
+private val unitShapeId = ShapeId.from("smithy.api#Unit")
+
+internal fun MemberShape.ofTypeUnit(): Boolean {
+    return this.target == unitShapeId
+}
```

Comment:
I lead you astray here. I think this can be replaced with the existing `isUnionShape` method of `MemberShape`.

Reply:
I am not sure if you did. I think you were right about suggesting we create a helper because I suspect `isUnionShape` does not check whether a member is of type `Unit` (and there does not seem to be `isUnitShape`). When I do replace `ofTypeUnit` with `isUnionShape`, the new unit test added in `UnionGeneratorTest.kt` no longer passes.

Reply:
Ok, sounds good then